lib/tasks/nfg_ui_tasks.rake

```diff
@@ -1,4 +1,53 @@
-# desc "Explaining what the task does"
-# task :nfg_ui do
-#   # Task goes here
-# end
+# This will copy the files in nfg_ui's `public/nfg_ui` folder
+# to the host app's `public` folder.
+Rake::Task['assets:precompile'].enhance do
+  Rake::Task['nfg_ui:copy_assets'].invoke
+end
+
+namespace :nfg_ui do
+  task :copy_assets => :"assets:environment"  do
+    puts 'rake nfg_ui:copy_assets has begun'
+    puts ''
+    asset_paths = Dir.glob(NfgUi::Engine.root.join('public', 'nfg_ui', '**', '*'))
+    asset_paths.each do |asset_path|
+      filename = File.basename(asset_path)
+      source_file = File.open(asset_path)
+      dest_file = File.join(Rails.root, 'public', filename)
+
+      # The file to be copied already exists. Defer to the host app
+      # and do not copy the file over.
+      if File.file?(dest_file)
+        puts 'File copy was unsuccessful'
+        puts "Reason: #{dest_file} already exists"
+        puts 'Source file:'
+        puts asset_path
+        puts ''
+
+      # The asset does not already exist!
+      else
+        # Copy the file over
+        FileUtils.copy_file source_file, dest_file, true
+
+        # The file was successfully copied
+        if File.file?(dest_file)
+          puts 'File copy was successful'
+          puts "Copied: #{asset_path} to #{dest_file}"
+          puts ''
+
+        # For reasons unknown, the file was not succesfully coppied
+        else
+          puts 'File copy was unsuccessful'
+          puts 'Reason: Unknown'
+          puts "Attempted command: `FileUtils.copy_file the_asset, dest_file_path, true`"
+          puts 'Source file:'
+          puts asset_path
+          puts 'Destination:'
+          puts Rails.root.join('public')
+          puts ''
+        end
+      end
+    end
+
+    puts 'rake nfg_ui:copy_assets has completed'
+  end
+end
```

Comment:
@pathakh can you double check me on this? Here's what I "think" I'm doing -- seeking a second set of eyes that my conditions are actually doing what they're supposed to.

_just a note: this rake task works and behaves as expected._

**Condition**:
```ruby
if File.file?(dest_file)
```
The host app (e.g.: Evo) already has a file with this filename in the folder **(though it does not have to be the exact same file or a variation of that file)** -- as in, I am not explicitly checking that the file in the host app's `public` directory is identical to the file to be copied over.

** Question #2 - File copying **
```
# asset_path is the full path including filename from Engine root... `/Users/jonathanroehm/projects/nfg_ui/public/nfg_ui/filename.ext`
source_file = File.open(asset_path)
dest_file = File.join(Rails.root, 'public', filename)
FileUtils.copy_file source_file, dest_file, true
```
Can you double check me on this:
1. For the `.copy_file`, the first arg, `source_file` in this instance is an actual `File` object (not a path). 
2. `dest_file` is the actual desired path of where to put this copied file.

If this isn't your cup-o-tea, no problem. Simply seeking a second set of eyes (code review) of this particular piece of functionality.